src/js/utils/dom-utils.js

```diff
@@ -92,6 +92,25 @@ function parseHTML(html) {
   return div;
 }
 
+function closest(el, selector, contentOnly=false) {
+  if (!isElementNode(el)) {
+    el = el.parentElement;
+  }
+  var matchesSelector = el.matches || el.webkitMatchesSelector || el.mozMatchesSelector || el.msMatchesSelector;
+
+  if (contentOnly && matchesSelector.call(el, selector)) {
+    return null;
+  }
+
+  while (el) {
+    if (matchesSelector.call(el, selector)) {
+      break;
+    }
+    el = el.parentElement;
+  }
+  return el;
+}
```

Comment:
It's probably very fast to traverse up the DOM tree like this, but it seems like it might be worthwhile to abort after we hit the editor element. An editor could be nested arbitrarily deeply in DOM. Passing in a final `stopElement` argument from the mutation handler would be nice.